views/js/runner/plugins/navigation/limitBackButton.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * Plugin that limits the use of the back button only when there is no responses in the current item.
+ * So, once a user has entered a response to any interaction, the back button will be disabled.
+ * We can check if an interaction is already answered by listening to `responsechange` on item runner.
+ *
+ * @author Ricardo Proença <ricardo@taotesting.com>
+ */
+define([
+    "jquery",
+    "lodash",
+    'taoTests/runner/plugin'
+], function ($, _, pluginFactory) {
+    'use strict';
+
+    var pluginName = "limitBackButton";
+
+    var defaultEventName = "change";
+
+    var customListeners = [
+        {
+            selector: ".widget-numpad",
+            eventName: "click"
+        }
+    ];
+
+    /**
+     * Gets the unique item key identifier for this plugin
+     *
+     * @param {string} itemIdentifier - item runner identifier
+     * @returns {string}
+     */
+    var getItemKey = function getItemKey(itemIdentifier) {
+        return itemIdentifier + "-limit-back-button";
+    };
+
+    /**
+     * Remove all listeners from an item
+     *
+     * @param {Object} testRunner - test runner, needed to find interactions elements
+     */
+    var removeAllListeners = function removeAllListeners(testRunner) {
+        _.forEach(customListeners, function (plugin) {
+            var $element = testRunner.getAreaBroker().getContainer().find(plugin.selector);
+
+            $element.off("." + pluginName);
+        });
+
+        var itemInteractions = testRunner.getAreaBroker().getContentArea().find('.qti-interaction');
```

Comment:
Please declare var on top to reflect hoisting.

Reply:
Right!